pkg/operation/botanist/vpnshoot.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+
+	"github.com/gardener/gardener/charts"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/vpnseedserver"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/vpnshoot"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+)
+
+// DefaultCoreDNS returns a deployer for the VPNShoot
+func (b *Botanist) DefaultVPNShoot() (vpnshoot.Interface, error) {
+	var (
+		image                  *imagevector.Image
+		err                    error
+		nodeNetwork            = b.Shoot.GetInfo().Spec.Networking.Nodes
+		ReversedVPNHeader      string
+		ReversedVPNEndPoint    string
+		ReversedVPNOpenVPNPort string
+		NodeNetworkCIDR        string
+	)
+	if nodeNetwork != nil {
+		NodeNetworkCIDR = *nodeNetwork
+	}
+	if b.Shoot.ReversedVPNEnabled {
+		image, err = b.ImageVector.FindImage(charts.ImageNameVpnShootClient, imagevector.RuntimeVersion(b.ShootVersion()), imagevector.TargetVersion(b.ShootVersion()))
+		if err != nil {
+			return nil, err
+		}
+		ReversedVPNHeader = "outbound|1194||" + vpnseedserver.ServiceName + "." + b.Shoot.SeedNamespace + ".svc.cluster.local"
+		ReversedVPNEndPoint = b.outOfClusterAPIServerFQDN()
+		ReversedVPNOpenVPNPort = "8132"
+	} else {
+		image, err = b.ImageVector.FindImage(charts.ImageNameVpnShoot, imagevector.RuntimeVersion(b.ShootVersion()), imagevector.TargetVersion(b.ShootVersion()))
```

Comment:
Instead of calling `FindImage` twice you could just save the correct image name in a variable and move the `FindImage` call down. I think this makes everything a bit more readable.